designer/fitting/dwipi.py

```diff
@@ -767,6 +771,126 @@ class object
         mkt = vectorize(mkt, self.mask)
         return mk, rk, ak, kfa, mkt, trace
 
+    def extractWMTI(self):
+        """Returns white matter tract integrity (WMTI) parameters. Warning:
+        this can only be run after fitting and DWI.extractDTI().
+
+        Parameters
+        ----------
+        (none)
+
+        Returns
+        -------
+        awf:        axonal water fraction
+        eas_ad:     extra-axonal space axial diffusivity
+        eas_rd:     extra-axonal space radial diffusivity
+        eas_tort:   extra-axonal space tortuosity
+        ias_ad:     intra-axonal space axial diffusivity
+        ias_rd:     intra-axonal space radial diffusivity
+        ias_tort:   intra-axonal space tortuosity
+        """
+        def wmtihelper(dt, dir, adc, akc, awf, adc2dt):
```

Comment:
It's usually not a great idea to have a function definition inside of a function definition. It's really on recommended for incredibly complex applications. Is there some reason you chose to do that?

Reply:
The only reason why this is inside a another function is to make it compatible with `joblib` so the parent function could parallelize the computations. While I can also do the same with this function outside, I don't want to add methods or functions that don't process an entire DWI. This function is used nowhere else and I don't want it to ever be visible to the end-user.

Reply:
I see. I suppose that's a compelling reason.